src/Core/IdentityServer/ClientStore.cs

```diff
@@ -106,6 +110,42 @@ public async Task<Client> FindClientByIdAsync(string clientId)
                     }
                 }
             }
+            else if (clientId.Contains("user."))
+            {
+                var idParts = clientId.Split('.');
+                if (idParts.Length > 2 && Guid.TryParse(idParts[2], out var id))
+                {
+                    var user = await _userRepository.GetByIdAsync(id);
+                    if (user != null)
+                    {
+                        var cliUris = new List<string>();
+                        for (var port = 8065; port <= 8070; port++)
+                        {
+                            cliUris.Add(string.Format("http://localhost:{0}", port));
+                        }
+
+                        return new Client
+                        {
+                            ClientId = clientId,
+                            RequireClientSecret = true,
+                            ClientSecrets = { new Secret(user.ApiKey.Sha256()) },
+                            AllowedScopes = new string[] { "api" },
+                            AllowedGrantTypes = GrantTypes.ClientCredentials,
+                            AccessTokenLifetime = 3600 * 1,
+                            RedirectUris = cliUris,
+                            PostLogoutRedirectUris = cliUris,
```

Comment:
Not sure we need these for ClientCredentials. These were used for SSO.

Reply:
Trashed